src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
-export interface Figure {
-
+export enum Shapes {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shapes;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shapes.Triangle;
+
+  constructor(
+    public color: Color,
+
+    public a: number,
+
+    public b: number,
+
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Error invalid input');
+    }
+
+    if (a >= b + c || b >= a + c || c >= b + a) {
+      throw new Error('Error invalid input');
+    }
+  }
 
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const square = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
+
+    return Math.round(square * 100) / 100;
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  shape = Shapes.Circle;
+
+  constructor(
+    public color: Color,
 
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Error radius <= 0');
+    }
+  }
+
+  getArea(): number {
+    return Math.floor(Math.PI * this.radius ** 2 * 100) / 100;
```

Comment:
Optional: you round 2 times over the code, better use the same logic for it.  You also could create a separate helper function - `rounder` for example for this purpose.